src/Voron/Data/Containers/Container.cs

```diff
@@ -331,14 +331,13 @@ public static List<long> GetAllIds(LowLevelTransaction llt, long containerId)
             return list;
         }
 
-        public static int GetEntriesInto(long containerId, ref int offset, Page page, Span<long> ids, int writingBufferOffset, out int itemsLeftOnCurrentPage)
+        public static int GetEntriesInto(long containerId, ref int offset, Page page, Span<long> ids, out int itemsLeftOnCurrentPage)
         {
-            var results = 0;
             if (page.IsOverflow)
             {
-                ids[results++] = page.PageNumber * Constants.Storage.PageSize;
+                ids[0] = page.PageNumber * Constants.Storage.PageSize;
                 itemsLeftOnCurrentPage = 0;
-                return results;
+                return 1;
             }
```

Comment:
This is as far as I could advance. I believe the main issue that @maciejaszyk is tracking is here. 

![image](https://user-images.githubusercontent.com/1045411/178385503-66faad8f-ad58-4fcd-81dc-45ed2df50885.png)

This results in this id been replicated. But it goes even further, this page is most likely not valid at all. 

cc @ayende

Reply:
I remember the case from last year, during the implementation of the deletion. Items despite being deleted still appeared with AllEntriesMatch. I think you could have removed the patch you made then.

Reply:
I don't recall this. What test / scenario is failing now?